pandas/core/internals/array_manager.py

```diff
@@ -287,7 +287,9 @@ def apply(
                 if not ignore_failures:
                     raise
                 continue
-            # if not isinstance(applied, ExtensionArray):
+            if not isinstance(applied, ExtensionArray):
+                if issubclass(applied.dtype.type, (str, bytes)):
+                    applied = np.array(applied, dtype=object)
```

Comment:
i think we ought to have a sanitize_str_dtypes or something akin to sanitize_to_nanoseconds

Reply:
This is now done as part of maybe_coerce_values, though that may be more heavy-weight than we want here